src/libudev/libudev-device-internal.h

```diff
@@ -23,16 +23,16 @@ struct udev_device {
         struct udev_device *parent;
         bool parent_set;
 
-        struct udev_list properties;
+        struct udev_list *properties;
         uint64_t properties_generation;
-        struct udev_list tags;
+        struct udev_list *tags;
         uint64_t tags_generation;
-        struct udev_list devlinks;
+        struct udev_list *devlinks;
         uint64_t devlinks_generation;
         bool properties_read:1;
         bool tags_read:1;
         bool devlinks_read:1;
-        struct udev_list sysattrs;
+        struct udev_list *sysattrs;
```

Comment:
<strike>Hmm, can you provide the justification for this change? `struct udev_device` is already "hidden", so by making those separate we're not "hiding" anything. And this increases the allocation cost quite a bit, because more memory is used (for the pointers), and a lot of small additional allocations are done. And this is a "hot path" in the sense that we have can thousands of such objects, so the efficiency of this data structure matters.</strike>

OK, so this is done ini preparation for the reimplementation using hashmaps. I think it'd be nice to say something in the commit message explaining this.